fedimint-server/src/config.rs

```diff
@@ -35,32 +35,47 @@ use crate::{ReconnectPeerConnections, TlsTcpConnector};
 const DEFAULT_MAX_CLIENT_CONNECTIONS: u32 = 1000;
 
 #[derive(Debug, Clone, Serialize, Deserialize)]
+/// All the serializable configuration for the fedimint server
 pub struct ServerConfig {
-    pub federation_name: String,
-    pub identity: PeerId,
-    pub hbbft_bind_addr: String,
-    pub api_bind_addr: String,
-    #[serde(with = "serde_tls_cert")]
-    pub tls_cert: rustls::Certificate,
+    /// Contains all configuration that needs to be the same for every server
+    pub consensus: ServerConfigConsensus,
+    /// Contains all configuration that is locally configurable and not secret
+    pub local: ServerConfigLocal,
+    /// Contains all configuration that is secret such as private key material
+    pub private: ServerConfigPrivate,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct ServerConfigPrivate {
     #[serde(with = "serde_tls_key")]
     pub tls_key: rustls::PrivateKey,
-
-    pub peers: BTreeMap<PeerId, Peer>,
     #[serde(with = "serde_binary_human_readable")]
     pub hbbft_sks: SerdeSecret<hbbft::crypto::SecretKeyShare>,
     #[serde(with = "serde_binary_human_readable")]
-    pub hbbft_pk_set: hbbft::crypto::PublicKeySet,
+    pub epoch_sks: SerdeSecret<hbbft::crypto::SecretKeyShare>,
+    pub modules: BTreeMap<String, serde_json::Value>,
+}
 
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct ServerConfigConsensus {
+    pub federation_name: String,
+    pub peers: BTreeMap<PeerId, Peer>,
     #[serde(with = "serde_binary_human_readable")]
-    pub epoch_sks: SerdeSecret<hbbft::crypto::SecretKeyShare>,
+    pub hbbft_pk_set: hbbft::crypto::PublicKeySet,
     #[serde(with = "serde_binary_human_readable")]
     pub epoch_pk_set: hbbft::crypto::PublicKeySet,
+    pub modules: BTreeMap<String, serde_json::Value>,
+}
 
-    // TODO split these into separate structs so they can be serialized independently
-    pub modules_local: BTreeMap<String, serde_json::Value>,
-    pub modules_private: BTreeMap<String, serde_json::Value>,
-    pub modules_consensus: BTreeMap<String, serde_json::Value>,
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct ServerConfigLocal {
+    pub identity: PeerId,
```

Comment:
One's identity inside the federation doesn't seem like something one could change oneself? All keys generated by the DKG are only valid for a certain peer id.

Reply:
I wanted consensus to be identical for all members.

Local config is modifiable, but might still fail if incorrect.  If you modified the identity you'd fail at config validation.

Reply:
Sounds like a good reason.